java/src/main/java/actorworkflow/CustomerLoyaltyWorkflow.java

```diff
@@ -0,0 +1,33 @@
+package actorworkflow;
+
+import io.temporal.workflow.QueryMethod;
+import io.temporal.workflow.SignalMethod;
+import io.temporal.workflow.WorkflowInterface;
+import io.temporal.workflow.WorkflowMethod;
+
+import java.util.ArrayList;
+
+// Workflow interface
+@WorkflowInterface
+public interface CustomerLoyaltyWorkflow {
+    @WorkflowMethod
+    String customerLoyalty(Customer customer);
+
+    @SignalMethod
+    void addLoyaltyPoints(int pointsToAdd);
+
+    @SignalMethod
+    void inviteGuest(Customer guest);
+
+    @SignalMethod
+    void ensureMinimumStatus(StatusTier status);
+
+    @SignalMethod
+    void cancelAccount();
+
+    @QueryMethod
```

Comment:
would just have single query that returns customer and then client can get this info from Customer object

Reply:
I initially went with this design choice based on the ideas behind the least-access principle: if the customer info object has -- or evolves to include -- name, email, address, guest lists, and more, why should I surface that data to a client that only needs the status information?

Granted, I have no clients written for this app, so perhaps your suggestion makes sense to keep things simple...

Reply:
if there are some specific requirements to expose different workflow state via different apis then having multiple query
is ok i think. same if for example you anticipate that the query response in some cases might take > blob size limit.

otherwise i think since queries are pretty expensive operations (especially on closed executions) would be just easier to return everything and do filtering on client side imo